tensorzero-optimizers/src/gepa/validate.rs

```diff
@@ -0,0 +1,1166 @@
+    // we construct it via serde (serialize HashMap, then deserialize to UninitializedChatTemplates)
+    let templates = {
+        let mut inner = HashMap::new();
+
+        // Get all explicit template names (filters out deprecated legacy templates with no schema)
+        let template_names = config.templates().get_all_explicit_template_names();
+
+        for template_name in template_names {
+            if let Some(template_with_schema) =
+                config.templates().get_named_template(&template_name)
+            {
+                // Create a new fake path with inline content to avoid duplicate path errors
+                // when multiple variants use the same template files
+                let fake_path = ResolvedTomlPathData::new_fake_path(
+                    format!("gepa_extracted/{template_name}"),
+                    template_with_schema.template.contents.clone(),
+                );
+
+                inner.insert(template_name, UninitializedChatTemplate { path: fake_path });
+            }
+        }
+        // Use serde to construct UninitializedChatTemplates from HashMap
+        // The #[serde(flatten)] attribute makes this work correctly
+        #[expect(clippy::expect_used)]
+        {
+            serde_json::from_value(
+                serde_json::to_value(&inner)
+                    .expect("Failed to serialize inner HashMap for UninitializedChatTemplates"),
+            )
+            .expect("Failed to deserialize UninitializedChatTemplates from serialized HashMap")
+        }
+    };
+
+    // Extract inference_params_v2 fields
+    let reasoning_effort = config.reasoning_effort().cloned();
+    let service_tier = config.service_tier().cloned();
+    let thinking_budget_tokens = config.thinking_budget_tokens();
+    let verbosity = config.verbosity().cloned();
+
+    UninitializedChatCompletionConfig {
+        weight: config.weight(),
+        model: config.model().clone(),
+        // Legacy fields - set to None since we only extract new-style templates
+        system_template: None,
+        user_template: None,
+        assistant_template: None,
+        input_wrappers: None,
+        // New-style templates
+        templates,
+        // Simple config fields
+        temperature: config.temperature(),
+        top_p: config.top_p(),
+        max_tokens: config.max_tokens(),
+        presence_penalty: config.presence_penalty(),
+        frequency_penalty: config.frequency_penalty(),
+        seed: config.seed(),
+        stop_sequences: config.stop_sequences().cloned(),
+        // Inference params v2 fields (flattened)
+        reasoning_effort,
+        service_tier,
+        thinking_budget_tokens,
+        verbosity,
+        // Other config
+        json_mode: config.json_mode().cloned(),
+        retries,
+        extra_body: config.extra_body().cloned(),
+        extra_headers: config.extra_headers().cloned(),
+    }
+}
```

Comment:
how hard would be be to implement this as `.into_uninitialized()` on the appropriate types? that might be better than keeping this logic here as a separarte thing. I don't think anything about this is GEPA-specific.